consensus/spos/consensusMessageValidator_test.go

```diff
@@ -1,3 +1,455 @@
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidHeaderSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidHeaderSize))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Header: []byte("header")}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithBlockBodyValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Header: []byte("header")}
+	err := wrk.CheckMessageWithBlockBodyValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockBodyValidity_InvalidBodySize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	bodyBytes := make([]byte, core.MegabyteSize+1)
+	_, _ = rand.Read(bodyBytes)
+	cnsMsg := &consensus.Message{Body: bodyBytes}
+	err := wrk.CheckMessageWithBlockBodyValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidBodySize))
+}
+
+func TestCheckMessageWithBlockBodyValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	bodyBytes := make([]byte, 100)
+	_, _ = rand.Read(bodyBytes)
+	cnsMsg := &consensus.Message{Body: bodyBytes}
+	err := wrk.CheckMessageWithBlockBodyValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithBlockBodyAndHeaderValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithBlockBodyAndHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockBodyAndHeaderValidity_InvalidBodySize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	bodyBytes := make([]byte, core.MegabyteSize+1)
+	_, _ = rand.Read(bodyBytes)
+	cnsMsg := &consensus.Message{Body: bodyBytes}
+	err := wrk.CheckMessageWithBlockBodyAndHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidBodySize))
+}
+
+func TestCheckMessageWithBlockBodyAndHeaderValidity_InvalidHeaderSize(t *testing.T) {
```

Comment:
You also can add a unit test for header size = 0

Reply:
done.